src/co2.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* exported getCo2eqPerByte */
+
+/**
+ * Environment-wide variable that will be overridden by the IIFE import.
+ */
+let co2Library;
+
+/**
+ * Imports the co2 library using eval().
+ *
+ * Note: eval() is a security risk, only used due to AS limitations.
+ */
+function importCo2Library() {
+  if (typeof co2Library !== 'undefined') return;
+  try {
+    eval(
+        UrlFetchApp
+            .fetch(
+                'https://cdn.jsdelivr.net/npm/@tgwf/co2@latest/dist/iife/index.js')
+            .getContentText());
+    co2Library = co2;
+  } catch (error) {
+    Logger.log(error);
+  }
+}
+
+/**
+ * Checks whether the given URL is hosted on a green hosting provider.
+ *
+ * @param {string} url The URL to check.
+ * @return {boolean} True if the URL is hosted on a green hosting provider,
+ *     false otherwise.
+ */
+function checkGreenHosting(url) {
+  if (typeof co2Library === 'undefined') {
```

Comment:
It's not clear to me why the library is imported here and in `getCo2eqPerByte()`

Reply:
It's not always importing the library on both functions, it's making sure it is imported whenever they are called. 

As these files aren't executed in their entirety, nor in order, nor there's another way to import dependencies (AppsScript limitations), what that function does is make sure that, if the reference to the library is not defined whenever these functions are invoked from the outside, it then fetches it.

If the library was already imported, it doesn't do anything. Happy to discuss over a quick conversation.